### PR TITLE
dnn: update dnn module to openvino version 2023/0

### PR DESCRIPTION
**Pull request by alimt1992**
dnn: update dnn module to openvino version 2023/0

    *git clone https://github.com/openvinotoolkit/openvino.git
    *git checkout -b releases/2023/0 origin/releases/2023/0
    *source <OPENVINO_INSTALLDIR>/setupvars.sh

### CODE REVIEW

**Comment by zongwave** [comment]
OpenVino 2023 released a Linux distribution package at https://docs.openvino.ai/2023.0/openvino_docs_install_guides_installing_openvino_from_archive_linux.html
With this release user can install & setup OpenVino environment by running a script “source /opt/intel/openvino_2023/setupvars.sh”, the library and include path also can be exported by openvino.pc file.

automake can use pkg-config tool to get OpenVino path:
$ pkg-config --cflags openvino
-DTBB_PREVIEW_WAITING_FOR_WORKERS=1 -DIE_THREAD=IE_THREAD_TBB -DOV_THREAD=OV_THREAD_TBB -I/opt/intel/openvino_2023.0.1/runtime/lib/intel64/pkgconfig/../../../..//runtime/include/ie -I/opt/intel/openvino_2023.0.1/runtime/lib/intel64/pkgconfig/../../../..//runtime/include


$ pkg-config --libs openvino
-L/opt/intel/openvino_2023.0.1/runtime/lib/intel64/pkgconfig/../../../..//runtime/lib/intel64 -lopenvino_onnx_frontend -lopenvino_paddle_frontend -lopenvino_pytorch_frontend -lopenvino_tensorflow_frontend -lopenvino_tensorflow_lite_frontend -lopenvino_c -lopenvino -ltbb -lpugixml

 
I think we can remove the compiling flags “OPENVINO_IE_LIBS_PATH” & “OPENVINO_IE_INC_PATH”.

**Comment by alimt1992** [comment]
> OpenVino 2023 released a Linux distribution package at https://docs.openvino.ai/2023.0/openvino_docs_install_guides_installing_openvino_from_archive_linux.html With this release user can install & setup OpenVino environment by running a script “source /opt/intel/openvino_2023/setupvars.sh”, the library and include path also can be exported by openvino.pc file.
> 
> automake can use pkg-config tool to get OpenVino path: $ pkg-config --cflags openvino -DTBB_PREVIEW_WAITING_FOR_WORKERS=1 -DIE_THREAD=IE_THREAD_TBB -DOV_THREAD=OV_THREAD_TBB -I/opt/intel/openvino_2023.0.1/runtime/lib/intel64/pkgconfig/../../../..//runtime/include/ie -I/opt/intel/openvino_2023.0.1/runtime/lib/intel64/pkgconfig/../../../..//runtime/include
> 
> $ pkg-config --libs openvino -L/opt/intel/openvino_2023.0.1/runtime/lib/intel64/pkgconfig/../../../..//runtime/lib/intel64 -lopenvino_onnx_frontend -lopenvino_paddle_frontend -lopenvino_pytorch_frontend -lopenvino_tensorflow_frontend -lopenvino_tensorflow_lite_frontend -lopenvino_c -lopenvino -ltbb -lpugixml
> 
> I think we can remove the compiling flags “OPENVINO_IE_LIBS_PATH” & “OPENVINO_IE_INC_PATH”.

changed dnn module Makefile.am accordingly. In order to access format_reader include and libs, an environment variable, "OPENVINO_EXTRAS_PATH" has been added.

File: modules/dnn/inference/Makefile.am
Comment (zongwave) [inline modules/dnn/inference/Makefile.am:1]: Please also remove $(OPENVINO_IE_LIBS_PATH) at line 50
libxcam_dnn_la_LDFLAGS = \
    $(XCAM_LT_LDFLAGS)   \
    -Wl,-rpath,$(OPENVINO_IE_LIBS_PATH) \
    $(NULL)

File: modules/dnn/inference/Makefile.am [hunk 1]
```diff
@@ -2,24 +2,22 @@ lib_LTLIBRARIES = libxcam_dnn.la
...
-    -lmyriadPlugin \
+    -L$(OPENVINO_EXTRAS_PATH)/bin/intel64/Release \
+    -lopenvino \
+    -lopenvino_intel_gpu_plugin \
+    -lopenvino_intel_cpu_plugin \
+    -lopenvino_intel_gna_plugin \
+    -lopenvino_hetero_plugin \
     -lformat_reader \
-    -lngraph \
+    -lie_samples_utils \
```
Comment (zongwave) [inline modules/dnn/inference/Makefile.am:20]: this library seems no used, try to remove it.

File: modules/dnn/inference/Makefile.am [hunk 2]
```diff
@@ -2,24 +2,22 @@ lib_LTLIBRARIES = libxcam_dnn.la
...
     $(top_builddir)/xcore/libxcam_core.la \
-    -L$(OPENVINO_IE_LIBS_PATH) \
-    -linference_engine_transformations \
-    -linference_engine \
-    -lclDNNPlugin \
-    -lMKLDNNPlugin \
-    -lGNAPlugin \
-    -lmyriadPlugin \
+    -L$(OPENVINO_EXTRAS_PATH)/bin/intel64/Release \
+    -lopenvino \
```
Comment (zongwave) [inline modules/dnn/inference/Makefile.am:14]: try to replace these -lopenvinoxxx by $(LIBOPENVINO_LIBS) generated by pkgconfig.

File: modules/dnn/inference/Makefile.am [hunk 3]
```diff
@@ -2,24 +2,22 @@ lib_LTLIBRARIES = libxcam_dnn.la
...
 XCAM_DNN_LIBS = \
     $(top_builddir)/xcore/libxcam_core.la \
-    -L$(OPENVINO_IE_LIBS_PATH) \
-    -linference_engine_transformations \
-    -linference_engine \
-    -lclDNNPlugin \
-    -lMKLDNNPlugin \
-    -lGNAPlugin \
-    -lmyriadPlugin \
+    -L$(OPENVINO_EXTRAS_PATH)/bin/intel64/Release \
```
Comment (zongwave) [inline modules/dnn/inference/Makefile.am:13]: move this line above -lformat_reader

**Review by zongwave — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -80,11 +80,15 @@ OpenCL is used to improve performance in different platforms.
...
         export GST_PLUGIN_PATH=/usr/local/lib/gstreamer-1.0:$GST_PLUGIN_PATH
+        export OPENVINO_EXTRAS_PATH=WORKDIR/openvino/
+        source /usr/local/setupvars.sh
 
     --prefix=/usr:
 
         export LD_LIBRARY_PATH=/usr/lib/:$LD_LIBRARY_PATH
         export GST_PLUGIN_PATH=/usr/lib/gstreamer-1.0:$GST_PLUGIN_PATH
+        export OPENVINO_EXTRAS_PATH=WORKDIR/openvino/
+        source /usr/setupvars.sh
```
Comment (zongwave) [inline README.md:91]: shell "setupvars.sh" are commonly used for many 3rd party libraries. put this openvino speicified shell under folder /usr might be not very friendly.

Comment (zongwave) [reply README.md:91]: same as line 91, for example:
/opt/intel/openvino_2023/setupvars.sh
/opt/intel/oneapi/setupvars.sh
... ...

**Review by zongwave — CHANGES_REQUESTED** [review]

**Review by zongwave — CHANGES_REQUESTED** [review]

**Review by zongwave — APPROVED** [review]

**Pull request merged**